### PR TITLE
Add Line schema

### PR DESCRIPTION
**Pull request by hand-dot**
ref: https://github.com/pdfme/pdfme/issues/329

### CODE REVIEW

File: packages/common/src/schema.ts [hunk 1]
```diff
@@ -50,13 +51,17 @@ export const Dict = z.object({
...
+  'schemas.line.color': z.string(),
 });
+export const Mode = z.enum(['viewer', 'form', 'designer']);
 
 export const Size = z.object({ height: z.number(), width: z.number() });
 
 export const Schema = z
   .object({
     type: z.string(),
+    readOnly: z.boolean().optional(),
```
Comment (hand-dot) [inline packages/common/src/schema.ts:64]: New properties for schema

**Comment by hand-dot** [comment]
Hey @peteward ,
I've included some refactoring as well, but essentially, 
I've added a new schema to `packages/schemas/src/line/index.ts`, and I would like you to mainly review that part.

File: packages/schemas/src/image/pdfRender.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ const getCacheKey = (schema: Schema, input: string) => `${schema.type}${input}`;
 
 export const pdfRender = async (arg: PDFRenderProps<ImageSchema>) => {
   const { value, schema, pdfDoc, page, _cache } = arg;
+  if (!value || !value.startsWith('data:image/')) return;
```
Comment (peteward) [inline packages/schemas/src/image/pdfRender.ts:9]: does this fix the extra request bug in #325?

File: packages/schemas/src/line/pdfRender.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    rotate,
+    position: { x, y },
+    opacity,
+  } = convertForPdfLayoutProps({ schema, pageHeight, applyRotateTranslate: false });
+  const pivot = { x: x + width / 2, y: y + height / 2 };
+  page.drawLine({
+    start: rotatePoint({ x, y: y + height / 2 }, pivot, rotate.angle),
+    end: rotatePoint({ x: x + width, y: y + height / 2 }, pivot, rotate.angle),
+    thickness: height,
+    color: hex2RgbColor(schema.color ?? '#ffffff'),
```
Comment (peteward) [inline packages/schemas/src/line/pdfRender.ts:20]: should white be the fallback colour or `DEFAULT_LINE_COLOR`?

**Review by peteward — APPROVED** [review]
Looks great. 

I don't have time try it out now, but the code all makes sense :+1:

**Pull request merged**